### PR TITLE
Update Dangerfile for noting tests, and stories on a PR

### PR DESCRIPTION
**Pull request by orta**
Adds the following:

* Any touched (new/modified) react components warn if they are not somewhere inside the storybook stories
* Any touched app files fails if there are no corresponding test files (not changes) - you need to specify that you are not writing tests in the PR body/title

---

WIP as I'm not sure it'll compile etc, I did it offline

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>Please assign someone to merge this PR, and optionally include people who should review.</td>
    </tr>
  </tbody>
</table>




<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Comment by orta** [comment]
Performance in tests is pretty terrible

screen shot 2017-03-21 at 10 12 08

I thought this was Danger's fault initially, but it isn't ([older pr](https://travis-ci.org/artsy/emission/builds/213155471)) - looks like a side-effect of the TS conversion.

**Comment by orta** [comment]
Bet this is the haste map generation - running tests twice "fixes it"

screen shot 2017-03-21 at 10 14 04

**Comment by alloy** [comment]
Can that be added to Travis’ cache?

**Comment by orta** [comment]
Yeah, they're in a determinate location, I've had to find it before

**Comment by orta** [comment]
Here we are: [cacheDirectory](http://facebook.github.io/jest/docs/configuration.html#cachedirectory-string)

File: dangerfile.ts [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+// Rules
+
+// When there are app-changes and it's not a PR marked as trivial, expect
+// there to be CHANGELOG changes.
+const changelogChanges = includes(modified, "CHANGELOG.md")
+if (modifiedAppFiles.length > 0 && !trivialPR && !changelogChanges) {
+  fail("No CHANGELOG added.")
+}
+
+// No PR is too small to warrent a paragraph or two of summary
```
Comment (alloy) [inline dangerfile.ts]: Nit: warrant

File: dangerfile.ts [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+// Warn if there are changes to package.json without changes to yarn.lock.
+const packageChanged = includes(modified, "package.json")
+const lockfileChanged = includes(modified, "yarn.lock")
+if (packageChanged && !lockfileChanged) {
+  const message = "Changes were made to package.json, but not to yarn.lock"
+  const idea = "Perhaps you need to run `yarn install`?"
+  warn(`${message} - <i>${idea}</i>`)
+}
+
+// Always ensure we assign someone, so that our Slackbot can do it's work correctly
```
Comment (alloy) [inline dangerfile.ts]: Nit: its

**Comment by alloy** [comment]
Good stuff 👌

**Comment by orta** [comment]
Nits picked, and made a separate PR for jest cache.

**Comment by orta** [comment]
This just needs to wait on 0.14.1 - in putting this in a real project I found a bug, after that - good to go

**Comment by orta** [comment]
Deployment is via CI, so is indeterminate - you know what that's like ;)

**Comment by alloy** [comment]
Ok, can you label this as WIP until that’s released?

**Comment by orta** [comment]
shipped ;)

**Comment by alloy** [comment]
Hah, I see now what you meant by “deployment is via CI” :D

**Comment by orta** [comment]
OK 👍 this is good to go

File: tsconfig.json [hunk 3]
```diff
@@ -6,6 +6,7 @@
         "rootDir": "src",
         "outDir": "build",
         "allowJs": true,
+        "pretty": true,
```
Comment (orta) [inline tsconfig.json]: oh yeah, this just provides some extra colours when you see typescript errors in the console.

**Pull request merged**